### PR TITLE
automate free scan functionality

### PR DESCRIPTION
**Pull request by mozrokafor**
# References: 
Jira: [MNTOR-2146](https://mozilla-hub.atlassian.net/browse/MNTOR-2146)

[MNTOR-2146]: https://mozilla-hub.atlassian.net/browse/MNTOR-2146?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Vinnl** [comment]
@mozrokafor Does the draft status of this PR mean you're not ready for reviews yet?

File: src/e2e/specs/landing.spec.js [hunk 1]
```diff
@@ -59,3 +59,54 @@ test.describe('Landing Page element verification', () => {
     })
   })
 })
+
+test.describe('Landing Page Functionality Verification', () => {
+  test.beforeEach(async ({ landingPage }) => {
+    await landingPage.open()
+  })
+
+  test('Verify landing page elements - free scan @smoke', async ({ landingPage, scanPage, page }) => {
```
Comment (Vinnl) [inline src/e2e/specs/landing.spec.js:68]: I thought old pages weren't going to be part of the smoke tests? (The landing page should be taken over by Bedrock.)

```suggestion
  test('Verify landing page elements - free scan', async ({ landingPage, scanPage, page }) => {
```

File: src/e2e/specs/landing.spec.js [hunk 2]
```diff
@@ -59,3 +59,54 @@ test.describe('Landing Page element verification', () => {
...
+
+      // verify redirect to sign in("/user/breaches")
+      expect(await scanPage.signUpForAlerts.getAttribute("href")).toBe("/user/breaches")
+      expect(await scanPage.getAlertsAboutBreachesButton.getAttribute("href")).toBe("/user/breaches")
+
+      // verify "have i been pwned" website redirect
+      expect(await scanPage.haveIBeenPwnedLink.getAttribute("href")).toBe("https://haveibeenpwned.com/")
+
+    }).toPass({
+      timeout: 2000
```
Comment (Vinnl) [inline src/e2e/specs/landing.spec.js:109]: :cry:

**Review by Vinnl — APPROVED** [review]
No real blockers I don't think.

File: src/e2e/specs/landing.spec.js (hunk 1 above)
Comment (mozrokafor) [reply src/e2e/specs/landing.spec.js:68]: ah ok, it was talked about but never heard any confirmations on this, will remove it from smoke

File: src/e2e/specs/landing.spec.js [hunk 3]
```diff
@@ -59,3 +59,54 @@ test.describe('Landing Page element verification', () => {
...
+    test.info().annotations.push({
+      type: "testrail",
+      description: "https://testrail.stage.mozaws.net/index.php?/cases/view/2255913",
+    });
+
+    // Intercept request and fulfill return
+    await page.route('**/api/v1/scan/', (route) => {
+         route.fulfill({
+          status: 200,
+            body: JSON.stringify({"success":true,"breaches":[],"total":0,"heading":"We found (mocked) <span class=\"breach-result-email\"></span> exposed in <span class=\"breach-result-count\"></span> data breaches.","dataClassStrings":[],"logos":[]}),
```
[the remark this answers is not in the document]
Comment (mozrokafor) [reply src/e2e/specs/landing.spec.js:79]: yea, creating a json file or class soon for these mocks

File: src/e2e/specs/landing.spec.js (hunk 2 above)
Comment (mozrokafor) [reply src/e2e/specs/landing.spec.js:109]: have another PR, with a more unified approach(still needs a hardcoded ms) hopefully helps.

File: src/e2e/specs/landing.spec.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Vinnl) [reply src/e2e/specs/landing.spec.js:79]: Hopefully we can remove this after Premium goes live (depending on how Bedrock's going to work) :crossed_fingers:

**Pull request merged**